pkg/operator/ceph/object/objectstore.go

```diff
@@ -625,19 +626,46 @@ func deletePools(context *Context, spec cephv1.ObjectStoreSpec, lastStore bool)
 	return nil
 }
 
+func allObjectPools(storeName string) []string {
+	baseObjPools := append(metadataPools, dataPoolName, rootPool)
+
+	poolsForThisStore := make([]string, 0, len(baseObjPools))
+	for _, p := range baseObjPools {
+		poolsForThisStore = append(poolsForThisStore, poolName(storeName, p))
+	}
+	return poolsForThisStore
+}
+
+func missingPools(context *Context) ([]string, error) {
+	// list pools instead of querying each pool individually. querying each individually makes it
+	// hard to determine if an error is because the pool does not exist or because of a connection
+	// issue with ceph mons (or some other underlying issue). if listing pools fails, we can be sure
+	// it is a connection issue and return an error.
+	existingPoolSummaries, err := cephclient.ListPoolSummaries(context.Context, context.clusterInfo)
+	if err != nil {
+		return []string{}, errors.Wrapf(err, "failed to determine if pools are missing. failed to list pools")
```

Comment:
I'm not sure why it wouldn't belong here. The method is called `missingPools`, so the context here is that the caller is looking for missing pools. At create pools, the additional context that can be added is `failed to find necessary pools (because pools necessary pools are missing)`.